### ISSUE TITLE
GLIBCXX_3.4.18

### ISSUE THREAD

**Issue by TR1D** [L-EARN/NoCheatZ-4#171]
NoCheatZ v4-r12, debian 8
``` 
failed to dlopen /home/steam/cs_awp/csgo/addons/NoCheatZ/nocheatzX.so error=/home/steam/cs_awp/bin/libstdc++.so.6: version `GLIBCXX_3.4.18' not found (required by /home/steam/cs_awp/csgo/addons/NoCheatZ/nocheatzX.so)
```

```
strings /usr/lib/i386-linux-gnu/libstdc++.so.6 | grep GLIBC
GLIBCXX_3.4
GLIBCXX_3.4.1
GLIBCXX_3.4.2
GLIBCXX_3.4.3
GLIBCXX_3.4.4
GLIBCXX_3.4.5
GLIBCXX_3.4.6
GLIBCXX_3.4.7
GLIBCXX_3.4.8
GLIBCXX_3.4.9
GLIBCXX_3.4.10
GLIBCXX_3.4.11
GLIBCXX_3.4.12
GLIBCXX_3.4.13
GLIBCXX_3.4.14
GLIBCXX_3.4.15
GLIBCXX_3.4.16
GLIBCXX_3.4.17
GLIBCXX_3.4.18
GLIBCXX_3.4.19
GLIBCXX_3.4.20
GLIBCXX_3.4.21
GLIBCXX_3.4.22
GLIBCXX_3.4.23
GLIBCXX_3.4.24
GLIBC_2.0
GLIBC_2.3
GLIBC_2.18
GLIBC_2.16
GLIBC_2.1
GLIBC_2.17
GLIBC_2.1.3
GLIBC_2.3.2
GLIBC_2.2
GLIBCXX_DEBUG_MESSAGE_LENGTH
```

**Comment by L-EARN** [comment]
Hi,

I doesn't matter which version of libstdc++ is installed on your system.
The game server ships with its own at /home/steam/cs_awp/bin/libstdc++.so.6

That's why I can't use newer compiler than gcc-4.8.

What's strange is the Travis CI builded r12 with g++-4.8 but introduced unused libcxx symbols which require glibcxx 3.4.18.
But the maximum version of gliccxx shipped with CSGO is 3.4.16.

Here are the versions the game server provides : 
```
Version definitions:
1 0x01 0x025f4d66 libstdc++.so.6
2 0x00 0x08922974 GLIBCXX_3.4
3 0x00 0x02297f81 GLIBCXX_3.4.1
        GLIBCXX_3.4
4 0x00 0x02297f82 GLIBCXX_3.4.2
        GLIBCXX_3.4.1
5 0x00 0x02297f83 GLIBCXX_3.4.3
        GLIBCXX_3.4.2
6 0x00 0x02297f84 GLIBCXX_3.4.4
        GLIBCXX_3.4.3
7 0x00 0x02297f85 GLIBCXX_3.4.5
        GLIBCXX_3.4.4
8 0x00 0x02297f86 GLIBCXX_3.4.6
        GLIBCXX_3.4.5
9 0x00 0x02297f87 GLIBCXX_3.4.7
        GLIBCXX_3.4.6
10 0x00 0x02297f88 GLIBCXX_3.4.8
        GLIBCXX_3.4.7
11 0x00 0x02297f89 GLIBCXX_3.4.9
        GLIBCXX_3.4.8
12 0x00 0x0297f860 GLIBCXX_3.4.10
        GLIBCXX_3.4.9
13 0x00 0x0297f861 GLIBCXX_3.4.11
        GLIBCXX_3.4.10
14 0x00 0x0297f862 GLIBCXX_3.4.12
        GLIBCXX_3.4.11
15 0x00 0x0297f863 GLIBCXX_3.4.13
        GLIBCXX_3.4.12
16 0x00 0x0297f864 GLIBCXX_3.4.14
        GLIBCXX_3.4.13
17 0x00 0x0297f865 GLIBCXX_3.4.15
        GLIBCXX_3.4.14
18 0x00 0x0297f866 GLIBCXX_3.4.16
        GLIBCXX_3.4.15
19 0x00 0x056bafd3 CXXABI_1.3
20 0x00 0x0bafd171 CXXABI_1.3.1
        CXXABI_1.3
21 0x00 0x0bafd172 CXXABI_1.3.2
        CXXABI_1.3.1
22 0x00 0x0bafd173 CXXABI_1.3.3
        CXXABI_1.3.2
23 0x00 0x0bafd174 CXXABI_1.3.4
        CXXABI_1.3.3
24 0x00 0x0bafd175 CXXABI_1.3.5
        CXXABI_1.3.4

Version References:
  required from libm.so.6:
    0x0d696910 0x00 35 GLIBC_2.0
  required from ld-linux.so.2:
    0x0d696913 0x00 33 GLIBC_2.3
  required from libgcc_s.so.1:
    0x09276060 0x00 36 GCC_4.2.0
    0x0d696910 0x00 34 GLIBC_2.0
    0x0b792653 0x00 32 GCC_3.3
    0x0b792650 0x00 29 GCC_3.0
  required from libc.so.6:
    0x0d696911 0x00 31 GLIBC_2.1
    0x0d696913 0x00 30 GLIBC_2.3
    0x09691f73 0x00 28 GLIBC_2.1.3
    0x09691972 0x00 27 GLIBC_2.3.2
    0x0d696910 0x00 26 GLIBC_2.0
    0x0d696912 0x00 25 GLIBC_2.2
```

At the time I dont really know if I missed something with the prebuilt libprotobuf or if Travis CI is making mistakes.

**Comment by L-EARN** [comment]
The 2 symbols requiring this glibcxx version was requested by the libprotobuf included in nocheatz.
It was built using c++11 and triggered the use of unordered_map instead of its own version. But the Ci was using a compiler a little bit newer than the one I had in debian.

Compiling libprotobuf with gnu++98 forced him to not use unordered_map and fixed the problem in r13.

```
Version References:
  required from libdl.so.2:
    0x0d696910 0x00 13 GLIBC_2.0
    0x0d696911 0x00 06 GLIBC_2.1
  required from libm.so.6:
    0x0d696910 0x00 07 GLIBC_2.0
    0x0d696911 0x00 04 GLIBC_2.1
  required from libc.so.6:
    0x0d696914 0x00 16 GLIBC_2.4
    0x06969197 0x00 15 GLIBC_2.17
    0x0d696912 0x00 14 GLIBC_2.2
    0x09691a74 0x00 12 GLIBC_2.2.4
    0x0d696911 0x00 09 GLIBC_2.1
    0x09691f73 0x00 05 GLIBC_2.1.3
    0x0d696910 0x00 03 GLIBC_2.0
  required from libstdc++.so.6:
    0x0297f860 0x00 17 GLIBCXX_3.4.10
    0x056bafd3 0x00 11 CXXABI_1.3
    0x0297f864 0x00 10 GLIBCXX_3.4.14
    0x02297f85 0x00 08 GLIBCXX_3.4.5
    0x08922974 0x00 02 GLIBCXX_3.4
```
